pkg/syncer/metadatasyncer.go

```diff
@@ -1041,6 +1043,15 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 
 		if err := metadataSyncer.volumeManager.DeleteVolume(ctx, volumeHandle, deleteDisk); err != nil {
 			log.Errorf("PVDeleted: Failed to delete disk %s with error %+v", volumeHandle, err)
+			return
+		}
+		if migrationFeatureEnabled && pv.Spec.VsphereVolume != nil {
+			// Delete the cnsvspherevolumemigration crd instance when PV is deleted
+			err = volumeMigrationService.DeleteVolumeInfo(ctx, volumeHandle)
+			if err != nil {
+				log.Errorf("PVDeleted: failed to delete volumeInfo CR for volume: %q. Error: %+v", volumeHandle, err)
```

Comment:
Say the workflow failed here, when do we retry deletion of the CR? 

Have we considered adding PV as an ownerRef in the CR? This way the CR gets automatically deleted when PV is deleted. Is there any reasons why we do not want to use ownerRef?

Reply:
In order to PV as an owner ref to CR, we need the PV name and PV UID. This information is not available in ControllerCreateVolume and ControllerPublishVolume because the request object only contains the volumeID/PV name.  UID of the parent object is mandatory to perform garbage collection correctly.

Reply:
> Say the workflow failed here, when do we retry deletion of the CR?

- If the workflow(only CR deletion) fails, we do not have a retry/clean-up in full sync.

- If the workflow for Volume Deletion fails, then we have a clean-up in full sync here: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/fullsync.go#L255

Reply:
>If the workflow(only CR deletion) fails, we do not have a retry/clean-up in full sync.

Can we fix it?

Reply:
Sure, addressed this in the latest commits

Reply:
Can you explain what the fix is? You may also want to update the PR description as well to reflect what you are doing in this PR. It helps in reviewing the code if it has changed significantly.

Reply:
@SandeepPissay @divyenpatel  I would like to keep the CR deletion/cleanup changes as a separate PR here: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/440

Reply:
Filed an issue to track the same: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/441